app/src/main/java/org/thoughtcrime/securesms/webrtc/NetworkChangeReceiver.kt

```diff
@@ -10,9 +11,18 @@ import org.session.libsignal.utilities.Log
 
 class NetworkChangeReceiver(private val onNetworkChangedCallback: (Boolean)->Unit) {
 
+    companion object {
+
+        // Method to check if a valid Internet connection is available or not
+        fun haveValidNetworkConnection(context: Context) : Boolean {
+            val cm = context.getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
+            return cm.activeNetwork != null
```

Comment:
The logic here is different from the one you manually added. Yours seems to be checking for more things. Are you sure that this is enough to actually test for network connection?

Also should this helper method live somewhere more generic, as this isn't specifically for the networkChangeReceiver? Is there a util class somewhere or a more appropriate location for this be be since it's more generic?

Reply:
Yup. That's what happens when you trust other people's code (i.e., it's what was in NetworkChangeReceiver's method). I've done some research and replace with an improved version of my original checking code.

I've also created a new `NetworkUtils` class and moved the method there, as nowhere else really seemed like a good fit.